cip-0010.md

```diff
@@ -0,0 +1,131 @@
+        CIP: 10
+        Title: Multi-Peer Multi-Asset sends "MPMA Sends"
+        Authors: John Villar (chiguireitor) & Javier Varona (pataegrillo)
+        Status: Draft
+        Type: Standards Track
+        Created: 2017-03-28
+        Discussions-To: https://counterpartytalk.org/t/cip0010-multi-peer-multi-asset-sends/2799
+
+# Abstract
+
+Establishes a new send message type geared towards batch processing to send multiple assets to multiple addresses.
+
+# Motivation
+
+* Simplify batch transactions
+* Reduce transaction fees
+
+# Rationale
+
+Multiple transactions are wasteful when the sender already knows who are the recipients and how much of each asset each one needs to receive. This functionality is especially useful for sites dealing with a group of asset to asset exchanges. MPMA addresses this issue by creating a lookup table for addresses and grouping sends by type of asset sent.
+
+# Definitions
+
+ * **LUT**: A lookup table. Elements are expressed as indexes into this table with the minimum amount of bits required for the total amount of elements in it.
+ * **RLE**: Run-length Encoding, a compression technique used to express multiple values grouped by type. A run-length specifies a group value and amount of elements contained in that group.
+ * **INT**: A big endian 8 byte integer,
+
+# Overview
+
+The main idea is to create a LUT for addreses so they can be expressed by simple indexes, and then compress each amount sent with simple RLE. This would save space by reducing the amount of bits dedicated to expressing the
+addresses. In the most extreme cases, reduction of bits needed to express each address is 168 times than
+standard sends.
+
+# Specification
+
+The message format is constructed in the following way (Where N is the number of recipients, M the number of assets, X the Log base 2 of N rounded up to the nearest integer):
+
+LUT size|LUT address 0|...|LUT address N|Asset Signaling bit 1|Asset SubList 0|...|Asset Signaling Bit 1|Asset Sublist M|Asset Signaling Bit 0|
+--------|-------------|---|-------------|---------------------|---------------|---|---------------------|---------------|---------------------|
+ 16 Bits|   21 bytes  |...|   21 bytes  |         1-bit       |   AssetList   |...|        1-bit        |   AssetList   |        1-bit        |
+
+`AssetList` is built in the following way:
+
+Asset Name|# Z of recipients|LUT Idx 0|Amount Idx 0|...|LUT Idx Z|Amount Idx Z|
+----------|-----------------|---------|------------|---|---------|------------|
+   INT    |     X bits      |  X bits |     INT    |...|  X bits |     INT    |
+
+Note that addresses are only 21 bytes. That's the result of decoding the base58 addresses into binary representation and stripping the last 4 bytes which are just a checksum.
```

Comment:
> That's the result of decoding the base58 addresses into binary representation and stripping the last 4 bytes which are just a checksum.

Are you suggesting to convert the base58 address into it's actual binary representation, or use the underlying RIPEMD-160 hash with network prefix?

Reply:
Yup, with network prefix without checksum like on enhanced sends